vendor/engines/saas/lib/saas/engine.rb

```diff
@@ -15,6 +15,8 @@ class Engine < ::Rails::Engine
         Analytics::Core.adapter = Segment::Analytics.new({
           write_key: ENV["SEGMENTIO_KEY"]
         })
+      else
+        Analytics::Core.adapter = NullAnalytics.new
```

Comment:
Rather than require that `adapter` be explicitly set, what if `Analytics::Core` initialized itself with a null or logging adapter by default?